index.js

```diff
@@ -1,42 +1,35 @@
 var os = require('os')
 var stats = require('./lib/stats')
+var platform = require('./lib/platform')
+var history = {}
 
-var wrapper = function (statType) {
+var wrapper = function (method) {
   return function (pid, options, cb) {
     if (typeof options === 'function') {
       cb = options
       options = {}
     }
 
-    return stats[statType](pid, options, cb)
-  }
-}
-
-var pusage = {
-  darwin: wrapper('ps'),
-  sunos: wrapper('ps'),
-  freebsd: wrapper('ps'),
-  netbsd: wrapper('proc'),
-  win: wrapper('win'),
-  linux: wrapper('proc'),
-  aix: wrapper('ps'),
-  unsupported: function (pid, options, cb) {
-    cb = typeof options === 'function' ? options : cb
+    if (method === platform.UNSUPPORTED) {
+      return cb(new Error(os.platform() + ' is not supported yet, please open an issue (https://github.com/soyuka/pidusage)'), null)
+    }
 
-    cb(new Error(os.platform() + ' is not supported yet, please fire an issue (https://github.com/soyuka/pidusage)'))
+    options.history = history
+    return stats[method](pid, options, cb)
   }
 }
 
-var platform = os.platform()
-platform = platform.match(/^win/) ? 'win' : platform // nor is windows a winner...
-platform = pusage[platform] ? platform : 'unsupported'
-
-exports.stat = function () {
-  pusage[platform].apply(stats, [].slice.call(arguments))
-}
+exports.stat = wrapper(platform)
 
 exports.unmonitor = function (pid) {
-  delete stats.history[pid]
+  if (!pid) {
+    for (var i in history) {
+      delete history[i]
+    }
+    return
+  }
+
+  delete history[pid]
 }
 
-exports._history = stats.history
+exports._history = history
```

Comment:
Why are we exposing this? It seems really dirty to me.

Reply:
For unit tests